doc/source/development/roadmap.rst

```diff
@@ -0,0 +1,145 @@
+.. _roadmap:
+
+=======
+Roadmap
+=======
+
+This page provides an overview of the major themes pandas' development. Implementation
+of these goals may be hastened with dedicated funding.
+
+Extensibility
+-------------
+
+Pandas Extension Arrays provide 3rd party libraries the ability to
+extend pandas' supported types. In theory, these 3rd party types can do
+everything one of pandas. In practice, many places in pandas will unintentionally
+convert the ExtensionArray to a NumPy array of Python objects, causing
+performance issues and the loss of type information. These problems are especially
+pronounced for nested data.
+
+We'd like to improve the handling of extension arrays throughout the library,
+making their behavior more consistent with the handling of NumPy arrays.
+
+String Dtype
+------------
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is Apache Arrow. Currently, Arrow provides
```

Comment:
This is not the most obvious actually. Rather a Dictionary encoded type (a supertype of Categorical for example), will actually be easier to do, and potentially a better fit for pandas as it can be mutable.

Reply:
A dictionary encoded type doesn't help for high-cardinality data, right? A dict-encoded string type will certainly give the same benefits of Categorical with semantics that are closer to a regular string dtype, but it will still lack performance high-cardinality cases.

I suppose the other contender would be [awkward-array](https://github.com/scikit-hep/awkward-array). I don't know of any other options.

If the specifics of using Arrow for memory storage of strings is contentious, we can write a version that's more generic and just says something like "Native support for text data".

Reply:
Very good distinction to make.

Updated in a549cf7b0662d719b39641312220285d98741c27

Reply:
> A dictionary encoded type doesn't help for high-cardinality data, right? A dict-encoded string type will certainly give the same benefits of Categorical with semantics that are closer to a regular string dtype, but it will still lack performance high-cardinality cases.
> 
> I suppose the other contender would be [awkward-array](https://github.com/scikit-hep/awkward-array). I don't know of any other options.
> 
> If the specifics of using Arrow for memory storage of strings is contentious, we can write a version that's more generic and just says something like "Native support for text data".

its pretty darn good for most real world cases though. This is the reason people use Categoricals now. I would argue that we should *always* use a DictionaryType in pandas for string data. Then *maybe* use say pyarrow to actually hold the strings themselves. I think this will be *much* *much* simpler to implement, preserve the api (mutation), and be less complex.